lib/tpm/key_attestation.rb

```diff
@@ -86,7 +86,13 @@ def trustworthy?
     def trust_store
       @trust_store ||=
         OpenSSL::X509::Store.new.tap do |trust_store|
-          root_certificates.each { |root_certificate| trust_store.add_cert(root_certificate) }
+          root_certificates.each do |root_certificate|
+            begin
+              trust_store.add_cert(root_certificate)
+            rescue OpenSSL::X509::StoreError => e
```

Comment:
Is it possible to filter duplicates by checking cert serial numbers?

Reply:
I mean

```diff
diff --git a/lib/tpm/key_attestation.rb b/lib/tpm/key_attestation.rb
index 4170f41..1b417cb 100644
--- a/lib/tpm/key_attestation.rb
+++ b/lib/tpm/key_attestation.rb
@@ -86,7 +86,7 @@ module TPM
     def trust_store
       @trust_store ||=
         OpenSSL::X509::Store.new.tap do |trust_store|
-          root_certificates.each { |root_certificate| trust_store.add_cert(root_certificate) }
+          root_certificates.uniq(&:serial).each { |root_certificate| trust_store.add_cert(root_certificate) }
         end
     end
 
```

Reply:
That's actually a good idea! I would suppose that the serial number is what identifies a certificate, right?

Reply:
According to [wikipedia](https://en.wikipedia.org/wiki/X.509#Certificates):

> In all versions, the serial number must be unique for each certificate issued by a specific CA (as mentioned in RFC 5280).